pkg/cmd/roachprod-microbench/compare.go

```diff
@@ -113,6 +133,35 @@ func (c *compare) readMetrics() (map[string]*model.MetricMap, error) {
 	return metricMaps, nil
 }
 
+func (c *compare) createComparisons(
+	metricMap model.MetricMap, oldID string, newID string,
+) map[string]map[string]*model.Comparison {
+
+	metricKeys := make([]string, 0, len(metricMap))
+	for sheetName := range metricMap {
+		metricKeys = append(metricKeys, sheetName)
+	}
+
+	comparisonMap := make(map[string]map[string]*model.Comparison, len(metricKeys))
+	for metricKey, metric := range metricMap {
+
+		// Compute comparisons for each benchmark present in both runs.
+		comparisons := make(map[string]*model.Comparison)
+		for name := range metric.BenchmarkEntries {
+			comparison := metric.ComputeComparison(name, oldID, newID)
+			if comparison != nil {
+				comparisons[name] = comparison
```

Comment:
This code is starting to look quite similar to what we have in `postToSlack`. I think we could combine these to use the same comparison logic.  `postToSlack` also sorts the results, with the highest delta first (based on `metric.Better` as well), which may be useful for the report generated by the threshold check.